functions/php-s3/terraform/main.tf

```
@@ -0,0 +1,85 @@
+variable "project_id" {
+  description = "Scaleway project ID"
+  type        = string
+}
+
+variable "access_key" {
+  description = "Scaleway access key"
+  type        = string
+}
+
+variable "secret_key" {
+  description = "Scaleway secret key"
+  type        = string
+  sensitive   = true
+}
+
+# Terraform provider
+terraform {
+  required_providers {
+    scaleway = {
+      source = "scaleway/scaleway"
+      version = ">= 2.8.0"
+    }
+  }
+  required_version = ">= 0.13"
+  backend "local" {
+    path = "state"
+  }
+}
+
+provider "scaleway" {
+  zone   = "fr-par-1"
+  region = "fr-par"
+}
+
+# S3 bucket
+resource "scaleway_object_bucket" "example_bucket" {
+  name = "php-s3-example"
+}
+
+# Function namespace and function
+resource "scaleway_function_namespace" "function_ns" {
+  project_id  = var.project_id
+  region      = "fr-par"
+  name        = "php-example-namespace"
+}
+
+resource "scaleway_function" "php_function" {
+  name           = "php-s3-function"
+  description    = "PHP example working with S3"
+  namespace_id   = scaleway_function_namespace.function_ns.id
+  runtime        = "php82"
+  handler        = "handler.run"
+  min_scale      = 0
+  max_scale      = 2
+  zip_file       = "../function/function.zip"
+  privacy        = "public"
+  deploy         = true
+
+  environment_variables = {
+    "S3_ENDPOINT" = "https://s3.fr-par.scw.cloud"
+    "S3_REGION" = "fr-par"
+    "S3_BUCKET" = "php-s3-example"
+  }
+
+  secret_environment_variables = {
+    "S3_ACCESS_KEY" = var.access_key
+    "S3_SECRET_KEY" = var.secret_key
+  }
+
+  timeouts {
+    create = "15m"
+    update = "15m"
+  }
+}
+
```

You could use Terraform to zip the function directly: 

```suggestion
data "archive_file" "function_archive" {
  type             = "zip"
  source_dir       = "${path.module}/../function/"
  output_file_mode = "0666"
  output_path      = "${path.module}/../function/function.zip"
```

And then add this to the function resource:

```
zip_file = data.archive_file.function_archive.output_path
zip_hash = data.archive_file.function_archive.output_sha
```
Stolen from here: https://github.com/scaleway/serverless-examples/blob/main/functions/redis-tls/terraform/main.tf

But a MakeFile is good also because it explains better how to zip :+1:

---

Good suggestion, I think I will switch to TF because otherwise we're making another assumption about the environment and existence of a `zip` command.

---

Thanks :relaxed: ! For even more Terraform goodness :candy:  you could generate an IAM S3 scoped API key with Terraform as well
Some examples:
Generating the key: https://github.com/cyclimse/severless-pypi/blob/main/deploying/terraform/iam.tf
And passing it to a container: https://github.com/cyclimse/severless-pypi/blob/main/deploying/terraform/main.tf#L57

---

I want to keep this as simple as possible, so will leave it for now but will remember for later!